flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/abilities/source/WatermarkPushDownSpec.java

```diff
@@ -131,4 +162,12 @@ public boolean equals(Object o) {
     public int hashCode() {
         return Objects.hash(super.hashCode(), watermarkExpr, idleTimeoutMillis);
     }
+
+    private long calculateIdleTimeoutMillis() {
+        long actualIdleTimeoutMillis = idleTimeoutMillis;
+        if (watermarkParams != null && watermarkParams.getSourceIdleTimeout() > 0) {
```

Comment:
What will happen if user want to set the idle timeout as zero?

Reply:
User may set idle timeout to zero actually,  I hive fixed it